### PR TITLE
add tools for querying job remaining time: `flux_job_timeleft(3)`, python `flux.job.timeleft()` and `flux-job timeleft`

### PR DESCRIPTION
**Pull request by grondo**
This WIP PR proposes a new `flux_job_timeleft(3)` function which fetches the remaining time for the current process, abstracting the implementation from callers because (as we've seen) the determination of _how_ to get the current time left is not exactly straightforward in all situations.

This function doesn't take a jobid but is still placed in libjob with the `flux_job` prefix since only jobs have a duration/expiration, so it doesn't seem to really belong under libflux.

Opening early as a WIP to get feedback (nothing about the interface is set in stone) and to allow @garlick to attempt addition of a `flux timeleft` utility, which can also act as a test vehicle for the function.

### CODE REVIEW

**Comment by grondo** [comment]
Removed WIP for an initial review. This PR now includes a python interface in `flux.job.timeleft()`. Possibly the last bit to do here is add a manpage for `flux_job_timeleft(3)`

**Comment by grondo** [comment]
Hold off. I found an issue with querying timeleft for pending jobs. I'll have to come back to that later today.

**Comment by grondo** [comment]
Ok, fixed the issue for pending jobs and added a brief man page. Ready for a review probably.

**Review by garlick — APPROVED** [review]
Nice work, couldn't spot anything wrong.

**Comment by grondo** [comment]
We should get approval from @ofaaland before merging probably. I'll request a review.

File: src/cmd/flux-job.c
Comment (chu11) [inline src/cmd/flux-job.c:3572]: maybe can be for a follow up PR, but perhaps option to output in plain ole seconds w/o the fsd suffix?  I can imagine users wanting to `sec=$(flux job timeleft)` and then immediately use the seconds in something else time related ... or possibly just check for `-eq 0`.  i.e. not waste time converting from fsd to a units they care about.

File: t/t2201-job-cmd.t
Comment (chu11) [inline t/t2201-job-cmd.t:776]: this test is in the ` testsuite: add tests for flux.job.timeleft() ` commit.  accidentally put here and not in the other commit with all of the other tests for `t2201-job-cmd`?

File: src/cmd/flux-job.c
Comment (grondo) [reply src/cmd/flux-job.c:3572]: Yeah, @garlick and I talked about that, but there didn't seem to be a known use case at this point and adding more development and test effort at this point seemed like a poor choice.

File: t/t2201-job-cmd.t
Comment (grondo) [reply t/t2201-job-cmd.t:776]: Ooh, good catch! Thanks, will fix

File: doc/man3/flux_job_timeleft.rst
Comment (ofaaland) [inline doc/man3/flux_job_timeleft.rst:17]: I see a comment below that says *timeleft is in units of seconds, but I don't see that specified in this man page.

File: src/cmd/flux-job.c
Comment (ofaaland) [reply src/cmd/flux-job.c:3572]: I don't know if there's an existing fsd <-> seconds convenience function.  Absent that, I personally would prefer seconds rather than fsd, because I can so easily convert seconds to minutes/hours/days but the reverse is not so easy.

That said, for SCR I'll be using python and using flux.job.timeleft(), so I have no concrete use case either.

**Review by ofaaland — COMMENTED** [review]
Great!  Thanks for taking care of this so quickly.  Looks good to me.  A couple of nits below.

File: src/cmd/flux-job.c
Comment (grondo) [reply src/cmd/flux-job.c:3572]: Ok, since there is two votes for seconds, I"ll go ahead and implement that, with a `-H, --human` option to return FSD.

What do people think should be returned for the `infinity` case: `-1` or `UINT_MAX`?

File: doc/man3/flux_job_timeleft.rst
Comment (grondo) [reply doc/man3/flux_job_timeleft.rst:17]: Great catch! Thanks, will fix.

File: src/cmd/flux-job.c
Comment (garlick) [reply src/cmd/flux-job.c:3572]: I'd vote for UINT_MAX.  Then a batch script could do something like

```
thing_runtime=3600
while test $(flux job timeleft) -gt $thing_runtime; do
    flux mini run -t $thing_runtime $thing
done
```

**Comment by grondo** [comment]
Thanks for the review @ofaaland! That was very helpful. I've made the following changes here:

 - `flux job timleft` now reports integral seconds left in the current or specified job's time limit. No time limit is reported with `ULONG_MAX`.
 - A `-H, --human` option was added to `flux job timeleft`
 - An entry for `timeleft` was added to `flux-job(1)` for completeness
 - The `flux_job_timeleft(3)` man page was updated to specify that the returned time left is in floating point seconds.

I squashed all the new stuff into the existing commits, hope that was ok.

**Comment by garlick** [comment]
All good here!

**Comment by grondo** [comment]
Ok, @ofaaland want to give an ACK? If so, then I'll set MWP.

**Comment by grondo** [comment]
Well, I think since @ofaaland already commented we can merge this one and get it out there. Setting MWP.

**Comment by ofaaland** [comment]
> Well, I think since @ofaaland already commented we can merge this one and get it out there. Setting MWP.

Shoot, sorry.  I was about to approve and then got distracted.  Thank you!

**Comment by grondo** [comment]
No problem! Thanks @ofaaland!

**Pull request merged**
